### ISSUE TITLE
Limit amount of CSS values we show in the style panel

### ISSUE THREAD

**Issue by kof** [webstudio-is/webstudio-designer#269]
In many cases CSS has a lot of non-practical values that would take up space in the UI. Should we create a mental framework that would help us reduce the supported lists of values?

Here is a proposal from @taylornowotny:

Webflow is the standard that we are going to beat. If Webflow has it, we should have it.

+ if there is feedback to include something additional, we consider that feedback
+ if we see a specific need to include something in addition, we can include that too
+ if some other app has a feature that is seen as useful, we can include that too

Can you think of a better way forward than this?

**Comment by taylornowotny** [comment]
This is basically the framework that we've been using to decide what to include in our UI, but it hasn't been brought to a discussion before.

Obviously it would be great to include every CSS property & value possible, but we don't want to overwhelm users with too many options when the majority of those options aren't commonly useful. By including only the most essential properties & values we are providing users with a more intuitive, more streamlined tool. Remember, Webstudio doesn't just have to be powerful, it also has to be easy for designers to use. 

So where is the sweet spot between simplicity and customization? That's the question of this topic, and my answer is: "Webflow +1"

In other words, let's include all the properties and values that Webflow has, plus anything else that we have a good reason to include. A good reason is a use-case that our users are likely to have. 

And remember, the full power of CSS is always available via custom code.

**Comment by rpominov** [comment]
How about something like this for dropdowns:

```
               ______________
display: | block                  |
              | inline                  |
              | other common values... |
              |     [more...]        |
              -----------------
```

A value selector isn't always a dropdown though.
But in general, what about hiding uncommon values / properties under some [more] or [expert settings] buttons, so they are not in the way but very easy to reach?

**Comment by kof** [comment]
@rpominov this is def. one option I was pointing out, we could potentially do it in 2 ways:

1. separate by a line, make the list size fit the manually selected important values, user won't see the rest until they scroll, visually it will look like there are no other values if they don't scroll, so not much visual overhead

2. like you said a "more" button

**Comment by okonet** [comment]
My take is: we should not necessarily mimic CSS in the designer but instead show and allow writing custom CSS somehow. This is coming from two goals:
1. Simplify visual editing and reduce the learning curve for newcomers esp. designers. For that we should not confront people with CSS terms since they are sometimes very confusing (ex.: flex-direction: column | row). 
1. Teach our audience CSS. By showing the resulting CSS we will enable a connection between visual design tools and code.
1. Allow advanced usages that aren't supported in UI (yet or maybe for good).

For 1: we already on a track to do so with the flex panel or box panel (for spacing) which resemble CSS values without the need to know their values upfront. 

What we are still missing is the CSS code editor that would allow for 2 and 3. 

WYT?

**Comment by kof** [comment]
@okonet full CSS support from code editor is definitely going to enable writing literally any CSS, with autocompletion that shows everything CSS has to offer.

The question is then if you want to be able to have an escape hatch to use an "unpopular" value directly from the list like this one

<img width="234" alt="Screenshot 2022-09-12 at 09 19 45" src="https://user-images.githubusercontent.com/52824/189595303-4c93ea35-55e5-4337-875f-d697f0286f84.png">

Besides the 2 options above there is actually a third option: an input field that lets you type.

What we also need to think about is how we are going to enable token values in menu/lists like that. Because this may be related.

**Comment by okonet** [comment]
Yeah I think we would need to figure out tokens first as it's related indeed: with design tokens we might 1. Use different names and 2. Have only a sub-set of values. 

That being said, design tokens not always match CSS properties. Some design tokens can represent a set of properties. A good example is typography where 1 token can represent multiple CSS properties.

**Comment by okonet** [comment]
To keep this discussion on topic, I'd say the easiest way is to sort values based on their usage frequency and alphabetically. This is what Chrome DevTools are doing as well.

**Comment by kof** [comment]
> sort values based on their usage frequency and alphabetically. This is what Chrome DevTools are doing as well.

Or semantically, based on how humans like to think about them mentally

**Comment by kof** [comment]
I think the biggest decisions we need to make right now in this regard is if we should manually shortlist the values we will display, because if we do, we need to build a separate dataset that will be fully manually maintained. I don't think we will be able to use the data generated from mdn for this.

**Comment by taylornowotny** [comment]
> My take is: we should not necessarily mimic CSS in the designer but instead show and allow writing custom CSS somehow. 

I agree. Custom CSS covers the edge cases while the main UI is focused on making the most common tasks as easy as possible. The visual UI doesn't need to support the entirety of CSS, and even if we wanted it to, we couldn't do so elegantly. As I mentioned in my message, having long lists of values or using a text input for every value would undermine the goal of making the UI easy and intuitive.

And as I discussed above, we have already been designing the Ui according to the "Webflow +1" paradigm. A fundamental shift from this would require a reworking of the entire design system and all of the style panel work that has been done so far.

**Comment by kof** [comment]
Display value list is reduced now, we will do the same for values as we add more visual controls for the sections
